wsmaster/che-core-api-workspace/src/test/java/org/eclipse/che/api/workspace/server/WorkspaceServiceTest.java

```diff
@@ -10,937 +10,941 @@
-  //                                                              .build();
-  //        return DtoConverter.asDto(config);
-  //    }
-  //
-  //    @Filter
-  //    public static class EnvironmentFilter implements RequestFilter {
-  //
-  //        public void doFilter(GenericContainerRequest request) {
-  //            EnvironmentContext.getCurrent().setSubject(new SubjectImpl(NAMESPACE, USER_ID, "token", false));
-  //        }
-  //    }
+  @SuppressWarnings("unused")
+  private static final ApiExceptionMapper MAPPER = new ApiExceptionMapper();
+
+  private static final String NAMESPACE = "user";
+  private static final String USER_ID = "user123";
+  private static final String API_ENDPOINT = "http://localhost:8080/api";
+  private static final Account TEST_ACCOUNT = new AccountImpl("anyId", NAMESPACE, "test");
+
+  @SuppressWarnings("unused")
+  private static final EnvironmentFilter FILTER = new EnvironmentFilter();
+
+  @Mock private WorkspaceManager wsManager;
+  @Mock private MachineTokenProvider machineTokenProvider;
+  @Mock private WorkspaceLinksGenerator linksGenerator;
+
+  private WorkspaceService service;
+
+  @BeforeMethod
+  public void setup() {
+    service = new WorkspaceService(API_ENDPOINT, wsManager, machineTokenProvider, linksGenerator);
+  }
+
+  @Test
+  public void shouldCreateWorkspace() throws Exception {
+    final WorkspaceConfigDto configDto = createConfigDto();
+    final WorkspaceImpl workspace = createWorkspace(configDto);
+    when(wsManager.createWorkspace(any(), anyString(), any())).thenReturn(workspace);
+
+    final Response response =
+        given()
+            .auth()
+            .basic(ADMIN_USER_NAME, ADMIN_USER_PASSWORD)
+            .contentType("application/json")
+            .body(configDto)
+            .when()
+            .post(
+                SECURE_PATH
+                    + "/workspace"
+                    + "?namespace=test"
+                    + "&attribute=stackId:stack123"
+                    + "&attribute=factoryId:factory123"
+                    + "&attribute=custom:custom:value");
+
+    assertEquals(response.getStatusCode(), 201);
+    assertEquals(
+        new WorkspaceImpl(unwrapDto(response, WorkspaceDto.class), TEST_ACCOUNT), workspace);
+    verify(wsManager)
+        .createWorkspace(
+            anyObject(),
+            eq("test"),
+            eq(
+                ImmutableMap.of(
+                    "stackId", "stack123",
+                    "factoryId", "factory123",
+                    "custom", "custom:value")));
+  }
+
+  @Test
+  public void shouldUseUsernameAsNamespaceWhenCreatingWorkspaceWithoutSpecifiedNamespace()
+      throws Exception {
+    final WorkspaceConfigDto configDto = createConfigDto();
+    final WorkspaceImpl workspace = createWorkspace(configDto);
+    when(wsManager.createWorkspace(
+            any(WorkspaceConfig.class), anyString(), anyMapOf(String.class, String.class)))
```

Comment:
with Java8, anyMapOf is not required, anyMap is enough for type inference

Reply:
thanks. I'll fix

Reply:
https://stackoverflow.com/questions/21783219/mockito-anymapof-nested-generics
Matchers.<Map<String, String >>any()

Reply:
BTW it's because I'm thinking of https://github.com/eclipse/che/pull/6719